osb-backup-manager/src/main/java/de/evoila/cf/backup/service/manager/BackupServiceManager.java

```diff
@@ -133,56 +135,63 @@ private void executeBackup(BackupExecutorService backupExecutorService, Endpoint
                 backupExecutorService.backup(endpointCredential, destination, id, item,
                         backupPlan.isCompression(), backupPlan.getPublicKey(), backupPlan.getIdAsString());
                 backupJob.setDestination(destination);
 
-                ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
                 CompletableFuture<AgentBackupResponse> completionFuture = new CompletableFuture<>();
-                completionFutures.add(completionFuture);
-                ScheduledFuture checkFuture = executor.scheduleAtFixedRate(() -> {
+                ScheduledFuture checkFuture = scheduledExcecutor.scheduleAtFixedRate(() -> {
                     try {
                         AgentBackupResponse agentBackupResponse = backupExecutorService.pollExecutionState(endpointCredential,
                                 "backup", id, new ParameterizedTypeReference<AgentBackupResponse>() {});
-
-                        updateWithAgentResponse(backupJob, item, agentBackupResponse);
-                        if (!agentBackupResponse.getStatus().equals(JobStatus.RUNNING)) {
-                            completionFuture.complete(agentBackupResponse);
+                        if (agentBackupResponse != null) {
+                            updateWithAgentResponse(backupJob, item, agentBackupResponse);
+                            if (!agentBackupResponse.getStatus().equals(JobStatus.RUNNING)) {
+                                completionFuture.complete(agentBackupResponse);
+                            }
                         }
                     } catch (BackupException ex) {
-                        completionFuture.complete(null);
+                        log.error("restore check failed, creating dummy response", ex);
+                        AgentBackupResponse dummyResponse = new AgentBackupResponse();
+                        dummyResponse.setStatus(JobStatus.FAILED);
+                        dummyResponse.setErrorMessage(ex.getMessage());
+                        completionFuture.complete(dummyResponse);
+                    } catch (Exception ex) {
+                        log.error("backup check failed", ex);
                     }
 
                 }, 0, 5, TimeUnit.SECONDS);
-                completionFuture.whenComplete((result, thrown) -> {
+                i++;
+                CompletableFuture<AgentBackupResponse> completionFutureWithCheck = completionFuture.whenComplete((result, thrown) -> {
                     if (result != null) {
                         if (result.getStatus().equals(JobStatus.SUCCEEDED)) {
                             backupJob.getFiles().put(item, result.getFilename());
                         }
 
                         updateWithAgentResponse(backupJob, item, result);
                     }
-
                     checkFuture.cancel(true);
-                    log.info("Finished execution of Backup Job");
```

Comment:
was the deletion of this log line on purpose?

Reply:
Yes, for the same reason as above

Reply:
this log line has been added before version 0.0.8

Reply:
Ok then we should keep it